students/Roy_Tate/lesson06/test_mailroom4.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/python3
+
+# Author/Student:  Roy Tate (githubtater)
+
+import unittest
+import mailroom4
+import os
+
+
+class MailroomTests(unittest.TestCase):
+    # build the initial donor dict (this is same initial dict as mailroom4.py)
+    donor_dict = {}
+    donor_dict['Bill'] = [1100, 2100, 3100]
+    donor_dict['Mark'] = [1000, 300000.23, 2000]
+    donor_dict['Jeff'] = [50, 450.355]
+    donor_dict['Roy'] = [4500, 7500, 11221, 30232, 323]
+
+    # Sum the donations and verify they exist in the donor report
+    def test_correct_sums_printed_in_report(self):
+        for donor, donations in self.donor_dict.items():
+            assert str(donor) in mailroom4.create_report()
+
+    # We want to ensure our signature appears as intended
+    # This also ensures that nothing appears after our signature
+    def test_verify_output_file_has_correct_signature(self):
+        output_file = 'Roy.txt'
+        signature = 'The good guys at the best organization'
+        with open(output_file, 'r') as f:
+            lines = f.readlines()
+            self.assertEqual(signature, lines[-1].strip())
```

Comment:
I don't see a mailroom function being called here? If I run this test I will not have Roy.txt file

also your test should first check that file exists, like `assert os.path.isfile`, ideally your asserts should fail not your test code